### PR TITLE
Add boolean filtering (#7190)

### PR DESCRIPTION
**Pull request by ad-elias**
filter-icp-true

Link to issue: https://github.com/twentyhq/twenty/issues/7190

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/object-filter-dropdown/components/ObjectFilterDropdownBooleanSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    setSelectedValue(value);
+    closeDropdown();
+  };
+
+  return (
+    <SelectableList
+      selectableListId="boolean-select"
+      selectableItemIdArray={options.map((option) => option.toString())}
+      hotkeyScope={RelationPickerHotkeyScope.RelationPicker}
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/object-record/object-filter-dropdown/components/ObjectFilterDropdownBooleanSelect.tsx:90]: Agree this is weird. Maybe you could create a followup issue for that we can take later? (as this was not introduced by your PR and is already in other places)

**Review by FelixMalfait — COMMENTED** [review]
Great work! Only issue I found is if you try to set the filter not from the filter menu but from the column, then you end up in a weird state 
Screenshot 2024-11-23 at 19 19 03

**Comment by FelixMalfait** [comment]
Screenshot 2024-11-23 at 20 08 09
Also this test (which is a bit annoying... Not sure if we should keep it tbh but let's keep it for now)

**Comment by ad-elias** [comment]
> Great work! Only issue I found is if you try to set the filter not from the filter menu but from the column, then you end up in a weird state Screenshot 2024-11-23 at 19 19 03

This is happening with all filters also in `main`. Let's fix it in a separate PR right after this one?

**Pull request merged**